pkgs/build-support/vm/default.nix

```diff
@@ -1322,6 +1322,19 @@ rec {
       archs = ["noarch" "x86_64"];
       packages = commonCentOSPackages ++ [ "procps-ng" ];
     };
+
+    centos74x86_64 = rec {
+      name = "centos-7.4-x86_64";
+      fullName = "CentOS 7.4 (x86_64)";
+      # N.B. Switch to vault.centos.org when the next release comes out
+      urlPrefix = http://mirror.centos.org/centos-7/7.4.1708/os/x86_64;
```

Comment:
I'm not sure what the NixPkg convention is for handling these sorts of things is.  When CentOS > 7.4 comes out, this URL is going to be moved to their permanent vault; and even though this upstream image is immutable they [haven't put it in the vault yet](http://vault.centos.org/7.4.1708/os/), because it's the most recent snapshot release.  

When it does move I can update the pkg expression, but older checkouts of NixPkgs will be broken.  Is that an OK compromise or should we let this PR sit until we have a vaulted url in ~1 year?

Reply:
I actually made this change as well during the weekend. 

I think it is fine to keep this url because a) its the only option currently and b) urls sometimes simply break and need to be replaced. We do not and can not expects resources to be forever available.